### PR TITLE
Make ucx-py installation blocks consistent

### PR DESCRIPTION
**Pull request by charlesbluca**
Consistently instructs users to use `pip install -v .` to install ucx-py, to remove any ambiguity from the instructions on disabling building with hwloc support.

### CODE REVIEW

File: docs/source/install.rst [hunk 1]
```diff
@@ -163,4 +163,6 @@ UCX-Py
 In UCX 1.10 and above, or for builds that don't need CUDA and InfiniBand support, users can disable building with hwloc support:
 
 ::
+
+    UCXPY_DISABLE_HWLOC=1 python setup.py build_ext --inplace
```
Comment (pentschev) [inline docs/source/install.rst:167]: I really don't understand the need for this. Why isn't just `UCXPY_DISABLE_HWLOC=1 pip install .` enough? I tested it locally just like that and it did work.

Comment (pentschev) [reply docs/source/install.rst:167]: Forgot to mention, but I also had another user testing it just the same and it worked for him as well.

Comment (charlesbluca) [reply docs/source/install.rst:167]: Ah I understand now that `pip install .` implicitly does the Cython build step, so we only need to pass the flag to that one command. Knowing that, this line is definitely unnecessary.

Out of interest, why in the [block directly above this](https://github.com/rapidsai/ucx-py/blob/branch-0.22/docs/source/install.rst#ucx-py) do we instruct to build the Cython separately from the `pip install`?

```
python setup.py build_ext --inplace
pip install .
```

Could this not just be `pip install -v .`?

Comment (pentschev) [reply docs/source/install.rst:167]: I'm not entirely sure, I think it was @quasiben who originally wrote it, it's possible he took the steps to build as he did when doing more verbose/debug builds. I'm in favor of removing that line and simplifying install steps. Could you do that @charlesbluca ?

Comment (charlesbluca) [reply docs/source/install.rst]: Sure!

**Review by pentschev — APPROVED** [review]
LGTM, thanks @charlesbluca !

File: docs/source/install.rst (hunk 1 above)
Comment (quasiben) [reply docs/source/install.rst]: I can't recall if I did this, and if I did, why it is the case.  +1 for removing

**Pull request merged**